<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="grid_folder" default="$(find sparse_map_utils)/maps/"/>
  <arg name="map_folder" default="$(find sparse_map_utils)/graphs/"/>
  <arg name="grid_name" default="$(arg grid_folder)bioroboticaV.yaml"/>
  <arg name="map_name" default="$(arg map_folder)bioroboticaV.txt"/>
  <node name="map_server" pkg="map_server" type="map_server" output="screen" args="$(arg grid_name)"/>

  <node name="sparse_server" pkg="sparse_map_server" type="sparse_map_server_node" output="screen">
    <param name="map_file" value="$(arg map_name)" type="str"/>
    <param name="k_neighboors" value="6" type="int"/>
    <param name="max_dist" value="1.0" type="double"/>
    <param name="min_dist" value="0.3" type="double"/>
    <param name="safety_height" value="1.5" type="double"/>
    <param name="safety_radius" value="0.3" type="double"/>
    <param name="connection_radius" value="0.20" type="double"/>
    <param name="validate_terminals" value="true"/>
    <param name="use_existing_graph" value="true" type="bool"/>
    <param name="nodes_file" value="$(find sparse_map_utils)/graphs/bioroboticaVGraph.pcd" type="str"/>
    <param name="edges_file" value="$(find sparse_map_utils)/graphs/bioroboticaVGraph.txt" type="str"/>
    <remap from="graph_marker" to="/hsrb/graph_marker"/>
    <remap from="make_plan" to="/sparse_map/make_plan"/>
  </node>

  <node name="simple_move" pkg="simple_move" type="simple_move_node" output="screen" args="--move_head --pot_fields">
    <remap from="simple_move/goal_path" to="sparse_plan"/>
    <remap from="/hardware/mobile_base/cmd_vel" to="/hsrb/command_velocity"/>

  </node>

  <node name="pot_fields" pkg="pot_fields" type="pot_fields_node" output="screen"/>

  <node name="loc_amcl" pkg="amcl" type="amcl" output="screen" args="scan:=/hsrb/base_scan">
    <param name="odom_alpha1" value="0.1"/>
    <param name="odom_alpha2" value="0.1"/>
    <param name="odom_alpha3" value="0.1"/>
    <param name="odom_alpha4" value="0.1"/>
    <param name="odom_alpha5" value="0.1"/>
    <!-- <param name="initial_pose_x" value="0.17"/> <param name="initial_pose_y" value="0.12"/> <param name="initial_pose_a" value="0.03"/> -->
    <param name="initial_pose_x" value="-1.14551"/>
    <param name="initial_pose_y" value="1.88108"/>
    <param name="initial_pose_a" value="-1.53454"/>

    <param name="update_min_a" value="0.1"/>
    <param name="laser_min_range" value="0.3"/>
    <param name="odom_model_type" value="omni-corrected"/>
  </node>

  <!-- <node name="fake_laser" pkg="fake_laser_scan" type="fake_laser_scan_node" output="screen"> <remap from="scan" to="/hsrb/base_scan"/> <param name="laser_frame" value="map"/> </node> -->

  <!-- <node name="fake_loc" pkg="fake_localization" type="fake_localization" output="scree"> <remap from="scan" to="/hsrb/base_scan"/> </node> -->
</launch>
